internal/storage/bucket_handle.go

```diff
@@ -477,14 +477,23 @@ func (b *bucketHandle) DeleteFolder(ctx context.Context, folderName string) (err
 	return err
 }
 
-func (b *bucketHandle) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (resp *control.RenameFolderOperation, err error) {
+func (b *bucketHandle) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (folder *gcs.Folder, err error) {
+	var controlFolder *controlpb.Folder
 	req := &controlpb.RenameFolderRequest{
 		Name:                "projects/_/buckets/" + b.bucketName + "/folders/" + folderName,
 		DestinationFolderId: destinationFolderId,
 	}
-	resp, err = b.controlClient.RenameFolder(ctx, req)
+	resp, err := b.controlClient.RenameFolder(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("longrunning operation: %w", err)
+	}
+
+	// Wait blocks until the long-running operation is completed,
+	// returning the response and any errors encountered.
+	controlFolder, err = resp.Wait(ctx)
+	folder = gcs.GCSFolder(b.bucketName, controlFolder)
 
-	return resp, err
+	return folder, err
```

Comment:
what is returned here, the destination foldername? if yes, can't we write a unit test?

Reply:
It is crashing at the point of resp.Wait(ctx), as we are mocking resp as nil.
Reason: control client's rename folder operation returns a [control.RenameFolderOperation](https://github.com/googleapis/google-cloud-go/blob/main/storage/control/apiv2/auxiliary.go#L30) response containing private data that we cannot access externally so we can not mock it in our unit tests.